### ISSUE TITLE
provider/aws: Support Elastic Beanstalk option settings with optional parameters

### ISSUE THREAD

**Issue by dharrisio** [hashicorp/terraform#6476]
In Elastic Beanstalk the aws:autoscaling:scheduledaction option settings require an additional parameter. After glancing through all the option settings, it looks like this is currently the only set of settings requiring more than namespace, name and value.

[http://docs.aws.amazon.com/elasticbeanstalk/latest/dg/command-options-general.html#command-options-general-autoscalingscheduledaction]

Thanks to @rurreac for bringing this issue up in #3157. I created this issue to track it separately from the old, closed issue. 

The fix for this looks fairly simple, as we may simply need to add an optional parameter in the setting schema. The hashing functions would also need to update to use the `resource_name` parameter in the hash when it is specified.

**Comment by WolfHazardous** [comment]
Hello @dharrisio, 
Do you have any information regarding the implementation of this missing feature?

**Comment by dharrisio** [comment]
@WolfHazardous I should be able to get to this sometime this week.

**Comment by WolfHazardous** [comment]
Hi @dharrisio ,

Waiting the implementation I've used a provisioner on the environment resource to achieve what I need.

```go
provisioner "local-exec" {
      command = "aws elasticbeanstalk update-environment --environment-id ${aws_elastic_beanstalk_environment.env.id} --cli-input-json file://${path.root}/beanstalk-update-env.json"
}
```

And the file beanstalk-options.json contains the required option with ResourceName : 
```json
{
    "OptionSettings": [
        {
            "ResourceName": "downweeknight",
            "Namespace": "aws:autoscaling:scheduledaction",
            "OptionName": "MinSize",
            "Value": "0"
        }...
```

As soon as the implementation will be publicly available I'll replace the hack :)

Cheers

**Comment by WolfHazardous** [comment]
Hi,

Waiting the implementation I've used a provisioner on the environment resource to achieve what I need.

```go
provisioner "local-exec" {
      command = "aws elasticbeanstalk update-environment --environment-id ${aws_elastic_beanstalk_environment.env.id} --cli-input-json file://${path.root}/beanstalk-update-env.json"
}
```

And the file beanstalk-options.json contains the required option with ResourceName : 
```json
{
    "OptionSettings": [
        {
            "ResourceName": "downweeknight",
            "Namespace": "aws:autoscaling:scheduledaction",
            "OptionName": "MinSize",
            "Value": "0"
        }...
```

As soon as the implementation will be publicly available I'll replace the hack :)

Cheers

**Comment by stack72** [comment]
Closed via #7376
